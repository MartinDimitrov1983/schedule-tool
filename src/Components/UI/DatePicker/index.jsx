import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@mui/material';
import { DatePicker as DefaultDatePicker } from 'react-date-picker';
import 'react-date-picker/dist/DatePicker.css';
import 'react-calendar/dist/Calendar.css';
import { DATE_DASH_FORMAT, DATE_DOT_FORMAT } from '../../../Constants';
import styles from './index.module.css';

const DatePicker = ({ title, onChange, date }) => {
  return (
    <Box
      sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}
    >
      <p>{title}</p>
      <DefaultDatePicker
        onChange={onChange}
        value={date}
        clearIcon={null}
        calendarIcon={null}
        dayPlaceholder={null}
        monthPlaceholder={null}
        yearPlaceholder={null}
        format={date ? DATE_DOT_FORMAT : DATE_DASH_FORMAT}
        className={styles.datePicker}
      />
    </Box>
  );
};

DatePicker.propTypes = {
  title: PropTypes.string,
  onChange: PropTypes.func,
  date: PropTypes.instanceOf(Date),
};

export default DatePicker;
