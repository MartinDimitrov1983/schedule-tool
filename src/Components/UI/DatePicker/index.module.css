.datePicker {
  width: 100%;
  min-height: 50px;
  text-align: center;
  border: 1px solid black;
}

.datePicker > div > div > input:invalid {
  background: white;
}
.datePicker > div > div > input:focus-visible {
  border: none;
}
