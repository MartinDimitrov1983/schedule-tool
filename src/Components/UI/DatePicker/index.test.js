import React from 'react';
import { render } from '@testing-library/react';

import DatePicker from './index';

test('Snapshot DatePicker', () => {
  const { asFragment } = render(
    <DatePicker title="Test" onChange={() => {}} date={null} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
