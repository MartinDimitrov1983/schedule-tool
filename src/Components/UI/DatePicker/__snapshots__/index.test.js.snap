// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot DatePicker 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-dvxtzn"
  >
    <p>
      Test
    </p>
    <div
      class="react-date-picker react-date-picker--closed react-date-picker--enabled datePicker"
    >
      <div
        class="react-date-picker__wrapper"
      >
        <div
          class="react-date-picker__inputGroup"
        >
          <input
            hidden=""
            max="275760-09-13"
            min="0001-01-01"
            name="date"
            style="visibility: hidden; position: absolute; z-index: -999;"
            type="date"
            value=""
          />
          <input
            autocomplete="off"
            class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
            data-input="true"
            inputmode="numeric"
            max="31"
            min="1"
            name="day"
            type="number"
            value=""
          />
          <span
            class="react-date-picker__inputGroup__divider"
          >
             
          </span>
          <input
            autocomplete="off"
            class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
            data-input="true"
            inputmode="numeric"
            max="12"
            min="1"
            name="month"
            type="number"
            value=""
          />
          <span
            class="react-date-picker__inputGroup__divider"
          >
             
          </span>
          <input
            autocomplete="off"
            class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
            data-input="true"
            inputmode="numeric"
            max="275760"
            min="1"
            name="year"
            step="1"
            type="number"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
